staging/src/k8s.io/apimachinery/pkg/api/validate/schema.go

```diff
@@ -46,6 +46,20 @@ func Required[T comparable](_ operation.Context, fldPath *field.Path, value, _ *
 	return field.ErrorList{field.Required(fldPath, "")}
 }
 
+// Forbidden verifies that the specified value is the zero-value for its
+// type.
+func Forbidden[T comparable](_ operation.Context, fldPath *field.Path, value, _ *T) field.ErrorList {
```

Comment:
Is there any types where the zero-value is non-trivial or the zero value is an actual value in the context of the field.      eg:  integer fields where 0 could be a meaningful input vs some defaulting?

Also if there is some defaulting done is there any concern on how this validator would interact?

Reply:
> Is there any types where the zero-value is non-trivial or the zero value is an actual value in the context of the field. eg: integer fields where 0 could be a meaningful input vs some defaulting?

Yes. Lots of detail about this here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

My goal here was to make `Forbidden` the direct opposite of the above `Required`.  I suspect we're going to need something more than what we have to cover zero-valued but not `omitempty` values you mentioned.

Reply:
Ah makes sense, thanks for the link. Sgtm!